src/analyze/analyze-chid.c

```diff
@@ -223,6 +252,65 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        int r;
+        const char *edid_content = NULL;
+        size_t edid_size = 0;
+        r = sd_device_get_sysattr_value_full(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
+
+        *ret_panel = new0(char16_t, 8);
+        if (!edid_get_panel_id(&header, *ret_panel))
+                return -EINVAL;
+        return 0;
+}
+
+static int edid_acquire(char16_t **ret_panel) {
+        assert(ret_panel);
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        int r, n = 0;
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_subsystem(e, "drm", true);
+        if (r < 0)
+                return r;
+
+        FOREACH_DEVICE(e, d) {
+                const char *drm_path = NULL;
+                r = edid_parse(d, ret_panel);
+                if (r == -ENOENT)
+                        continue;
+                if (r < 0) {
+                        r = sd_device_get_syspath(d, &drm_path);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get DRM device path: %m");
+                        log_warning("Skipping DRM device with invalid EDID: %s", drm_path);
+                        continue;
+                }
+                n++;
+                if (n > 1) {
+                        log_error("Detected multiple monitors, please specify a sysfs path to the EDID blob");
+                        return -EEXIST;
+                }
+        }
+
+        return 0;
```

Comment:
we want that all our functions either log about all errors or about none (in which case the caller can log). (that said: debug logging is always ok)

this funcion here logs about some errors, but not about others, which is not ok.

i think it makes sense to not log here, and let the caller log.

i'd return -ENOTUNIQ or so in case there are multiple devices, so that the caller can generate a nice warning